### PR TITLE
expose idp_metadata_source info in `oxide silo idp saml create`

### LINKED ISSUE #162: Can't provide IDP metadata
Similar to https://github.com/oxidecomputer/oxide.rs/issues/161, there's no way to pass `idp_metadata_source` with `oxide silo idp saml create`

```console
$ oxide silo idp saml create --help
Create a SAML IdP

Usage: oxide silo idp saml create [OPTIONS] --silo <silo> --acs-url <acs-url> --description <description> --idp-entity-id <idp-entity-id> --name <name> --slo-url <slo-url> --sp-client-id <sp-client-id> --technical-contact-email <technical-contact-email>

Options:
      --silo <silo>
          Name or ID of the silo
      --acs-url <acs-url>
          service provider endpoint where the response will be sent
      --description <description>

      --group-attribute-name <group-attribute-name>
          If set, SAML attributes with this name will be considered to denote a user's group membership, where the attribute value(s) should be a comma-separated list of group names.
      --idp-entity-id <idp-entity-id>
          idp's entity id
      --name <name>

      --slo-url <slo-url>
          service provider endpoint where the idp should send log out requests
      --sp-client-id <sp-client-id>
          sp's client id
      --technical-contact-email <technical-contact-email>
          customer's technical contact for saml configuration
  -h, --help
          Print help
```

### PR DESCRIPTION
**Pull request by ahl**
If we like this, I'd like to move the API and progenitor towards auto-generating it, but doing this one off to see if we **do** like it.

Closes #162

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -55,6 +55,29 @@ impl<'a> Tree<'a> {
...
+                    .arg(
+                        clap::Arg::new("metadata-url")
+                            .long("metadata-url")
+                            .value_name("url")
+                            .value_parser(clap::value_parser!(String)),
+                    )
+                    .arg(
+                        clap::Arg::new("metadata-value")
+                            .long("metadata-value")
+                            .value_name("xml")
```
Comment (jmpesp) [inline cli/src/main.rs:71]: should we note that this is base64 encoded xml?

**Review by jmpesp — APPROVED** [review]
Looks good!

File: cli/src/main.rs [hunk 2]
```diff
@@ -55,6 +55,29 @@ impl<'a> Tree<'a> {
...
+                            .value_parser(clap::value_parser!(String)),
```
Comment (luqmana) [inline cli/src/main.rs:72]: Given how long this value can get (it is XML after all), should this take a file instead? If one really wants to pass it as a string they can pipe it to `stdin` with `-` (there's the `<( echo $XML )` redirect syntax but was that just a bash-ism?).

Comment (ahl) [reply cli/src/main.rs:72]: so: yeah! however, my primary goal was to make this stuff 1. possible and 2. generateable from progenitor. Could I defer this convenience (which I agree would be much nicer to use).

Comment (luqmana) [reply cli/src/main.rs:72]: sure! it is an improvement over the status quo

**Pull request merged**